roles/display_deployment_plan/templates/plan.j2

```diff
@@ -0,0 +1,41 @@
+An OpenShift cluster is about to be deployed. Please double check the provided inventory details.
+
+* General configuration
+
+    {{ row_format_str | format('Cluster name', inventory.cluster_name) }}
+    {{ row_format_str | format('Base DNS domain', inventory.base_dns_domain) }}
+
+    {{ row_format_str | format('API Virtual IP', inventory.api_vip) }}
+    {{ row_format_str | format('Ingress Virtual IP', inventory.ingress_vip) }}
+
+    {{ row_format_str | format('OpenShift version', inventory.openshift_full_version) }}
+
+* Prerequisite services configuration
+
+    {{ row_format_str | format('Setup NTP Service', inventory.setup_ntp_service) }}
+    {{ row_format_str | format('Setup DNS Service', inventory.setup_dns_service) }}
+    {{ row_format_str | format('Setup Registry Service', inventory.setup_registry_service) }}
+    {{ row_format_str | format('Setup HTTP Store Service', inventory.setup_http_store_service) }}
+
+* Groups and hosts configuration
+
+  {% for group_name in inventory.groups_filtered %}
+  {{group_name}}:
+  {% for host_name in groups[group_name] %}
+  {% if hostvars[host_name]['vendor'] is defined and hostvars[host_name]['vendor'] == 'KVM'  %}
+  {{ hosts_row_format_str | format(host_name, ((hostvars[host_name]['ansible_host'] +' (' + (hostvars[host_name]['vm_host'] | default(messages.value_missing)) + ')' ) | default(messages.value_missing))) }}
+  {% else %}
+  {{ hosts_row_format_str | format(host_name, (hostvars[host_name]['ansible_host'] | default(messages.value_missing))) }}
+  {% endif -%}
```

Comment:
I think it'd be possible to simplify this a tiny bit. We could create a new variable for rendering `ansible_host`, and push string concatenations out of filter invocations.

First, I'd add a dedicated formatting string for all rows extended with extra information (e.g. to display the `vm_host` assignment):
```yaml
hosts_row_extended_format_str: '%-{{ (left_column_width - hosts_row_padding) }}s %s (%s)'
```
Notice that the string has the extra `(%s)` at the end to display the info in brackets, so we don't have to do additional string concatenations anymore. 

Next, the expression for rendering `ansible_host` is being repeated here, and it gives us a chance to turn it into a dedicated variable. 

I think that the `if` condition can be simplified too. Removing the check for the `KVM` vendor will allow the user to catch errors with an invalid inventory config, e.g. baremetal nodes with assigned `vm_host`.

An updated snippet would look like the following:
```yaml
{%- set host_ansible_host = hostvars[host_name]['ansible_host'] | default(messages.value_missing) -%}
{% if hostvars[host_name]['vm_host'] is defined %}
{{ hosts_row_extended_format_str | format(host_name, host_ansible_host, hostvars[host_name]['vm_host']) }}
{% else %}
{{ hosts_row_format_str | format(host_name, host_ansible_host) }}
{% endif -%}
```